### PR TITLE
Make Gubernator github hook processing more memory-efficient.

### PR DESCRIPTION
**Pull request by rmmh**
Bail out entirely if it's unlikely we'll succeed.

### CODE REVIEW

File: gubernator/github/classifier.py [hunk 1]
```diff
@@ -62,13 +65,25 @@ def classify_issue(repo, number):
...
-    logging.debug('classifying %s %s (%d events)', repo, number, len(events))
+    event_keys = list(models.GithubWebhookRaw.query(ancestor=ancestor).fetch(keys_only=True))
+
+    logging.debug('classifying %s %s (%d events)', repo, number, len(event_keys))
+    event_tuples = []
+    last_event_timestamp = datetime.datetime(2000, 1, 1)
+
+
+    if len(event_keys) > 800:
+        logging.error('too many events. blackholing.')
```
Comment (spxtr) [inline gubernator/github/classifier.py]: Should this be an error or a warning?

**Comment by spxtr** [comment]
I'm fine with this change, but I'm worried it'll start happening again, especially if little optimizations like deduping strings makes such a big difference.

Is there some way to easily see the drop rate?

/lgtm

**Comment by rmmh** [comment]
Github-side dropping isn't easy, but you can write a query to find every time an instance gets killed from using too much memory-- which is the most common cause of dropped hooks.

**Comment by spxtr** [comment]
Sounds fine.

**Pull request merged**